File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        difference  = {}
        result = []
        for idx in range(len(nums)):
            diff = target - nums[idx]
            if nums[idx] in difference:
                return [difference[nums[idx]], idx]
            difference[diff] = idx
        return []

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> remaining = new HashMap<>();
        for(int index = 0; index < nums.length; index++){
            int diff = target - nums[index];
            if(remaining.containsKey(diff)){
                result[0] = remaining.get(diff);
                result[1] = index;
            }else{
                remaining.put(nums[index], index);
            }
        }
        return result;
    }
}